packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -246,25 +247,18 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
       const videoOptions = { localVideoStreams: this.localStream ? [this.localStream] : undefined };
 
       const isTeamsMeeting = !('groupId' in this.locator);
-
-      if (isTeamsMeeting) {
-        this.call = this.callAgent.join(this.locator as TeamsMeetingLinkLocator, {
-          audioOptions,
-          videoOptions
-        });
-      } else {
-        this.call = this.callAgent.join(this.locator as GroupCallLocator, {
-          audioOptions,
-          videoOptions
-        });
-      }
-
-      this.context.setCallId(this.call.id);
-      // Resync state after callId is set
-      this.context.updateClientState(this.callClient.getState());
-      this.handlers = createDefaultCallingHandlers(this.callClient, this.callAgent, this.deviceManager, this.call);
-      this.subscribeCallEvents();
-      return this.call;
+      const call = isTeamsMeeting
+        ? this.callAgent.join(this.locator as TeamsMeetingLinkLocator, {
+            audioOptions,
+            videoOptions
+          })
+        : this.callAgent.join(this.locator as GroupCallLocator, {
+            audioOptions,
+            videoOptions
+          });
+
+      this.updateContextAfterJoiningCall(call);
```

Comment:
Naming nit: This method does more than upate `this.context`. How about `processNewCall`, or `setupNewCall`

Reply:
I like processNewCall -- updated